src/libraries/System.Runtime/tests/System/DecimalTests.cs

```diff
@@ -788,6 +788,7 @@ public static IEnumerable<object[]> Parse_Valid_TestData()
             yield return new object[] { "  123  ", defaultStyle, null, 123m };
             yield return new object[] { (567.89m).ToString(), defaultStyle, null, 567.89m };
             yield return new object[] { (-567.89m).ToString(), defaultStyle, null, -567.89m };
+            yield return new object[] { "0.6666666666666666666666666666500000000000000000000000000000000000000000000000000000000000000", defaultStyle, null, 0.6666666666666666666666666666m };
```

Comment:
This was returning `0.6666666666666666666666666667m`, is that correct?

Reply:
Nope. It was returning `0.6666666666666666666666666666m`, the correct value. This is just a good test case though where `Decimal.MaxDigits == 30`, so we can test the `if (digCount < maxDigCount)` condition fully.